src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java

```diff
@@ -0,0 +1,235 @@
+        // bugzilla 2372 set up a seraching mode so the next and previous action will
+        // know
+        // what to do
+
+        if (YES.equals(request.getParameter("search"))) {
+
+            request.setAttribute(IN_MENU_SELECT_LIST_HEADER_SEARCH, "true");
+        }
+
+        return providers;
+    }
+
+    @Override
+    protected String getDeactivateDisabled() {
+        return "false";
+    }
+
+    @Override
+    protected String getAddDisabled() {
+        return "true";
+    }
+
+    @Override
+    protected String getEditDisabled() {
+        return "true";
+    }
+
+    @Override
+    protected int getPageSize() {
+        return SystemConfiguration.getInstance().getDefaultPageSize();
+    }
+
+    // gnr: Deactivate not Delete
+    @PostMapping(value = "/DeleteProvider")
+    public ResponseEntity<String> showDeleteProvider(HttpServletRequest request,
+            @RequestParam(value = ID, required = false) @Pattern(regexp = "[a-zA-Z0-9 -]*") String id,
+            @Valid @ModelAttribute("form") ProviderMenuForm form, BindingResult result)
+            throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        if (result.hasErrors()) {
+            // redirectAttributes.addFlashAttribute(Constants.REQUEST_ERRORS, result);
+            // findForward(FWD_FAIL_DELETE, form);
+            return ResponseEntity.badRequest().body(VALIDATION);
+        }
+
+        String[] IDs = id.split(",");
+        List<String> selectedIDs = new ArrayList<>();
+        for (int i = 0; i < IDs.length; i++) {
+            selectedIDs.add(IDs[i]);
+        }
+        List<Provider> providers = new ArrayList<>();
+        for (int i = 0; i < selectedIDs.size(); i++) {
+            Provider provider = new Provider();
+            provider.setId(selectedIDs.get(i));
+            provider.setSysUserId(getSysUserId(request));
+            providers.add(provider);
+        }
+
+        try {
+            providerService.deactivateProviders(providers);
+            return ResponseEntity.ok("Providers successfully deactivated.");
+        } catch (LIMSRuntimeException e) {
+            // bugzilla 2154
+            LogEvent.logError(e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error while deleting providers.");
+
+            // String errorMsg;
+            // if (e.getCause() instanceof org.hibernate.StaleObjectStateException) {
+            //     errorMsg = "errors.OptimisticLockException";
+            // } else {
+            //     errorMsg = "errors.DeleteException";
+            // }
+            // result.reject(errorMsg);
+            // redirectAttributes.addFlashAttribute(Constants.REQUEST_ERRORS, result);
+            // return findForward(FWD_FAIL_DELETE, form);
+
+        }
+        // redirectAttributes.addAttribute(FWD_SUCCESS, true);
+        // return findForward(FWD_SUCCESS_DELETE, form);
```

Comment:
likewise here!. just no comment them out. try as much as possible to keep the initial functionality intact

Reply:
this is an unreachable code